addons-tutorial.md

```diff
@@ -0,0 +1,91 @@
+Addons are used to customize the images used for the containers. An addon has *hooks* that are
+invoked at various points in the lifecycle of a container. A good example is an addon which
+enables the SSH service inside a container to provide access from the outside for any kind
+of automation of the container.  
+
+Addons are created independently, and can be attached to individual applications.
+
+<h2 id='heading--build-your-own-addon'>Build your first addon</h2>
+
+Any addon must have a `manifest.yaml` file and at least one hook in the `hooks` directory, e.g:
+
+```text
+manifest.yaml
+hooks/pre-start
+```
+### Metadata
+The `manifest.yaml` file contains metadata information about your addon.
+Create a  `manifest.yaml` file that contains the following
+
+```yaml
+name: ssh
+description: |  
+	Enable SSH access when starting a container  
+```
+
+### Hooks
+The `hooks` directory contains executable files where each corresponds to a single hook. Hooks can be implemented in any language, but we are using a bash script here.
+
+Create a directory named `hooks` and, in it, a file named `pre-start` containing the following script:
+```bash
+#!/bin/bash
+
+systemctl start sshd
+```
+
+> INFO: supported hooks are: `pre-start`, `post-start` and `post-stop`. Read more about them [here](TODO: INSERT LINK TO ADDON REFERENCE)
+
+### Create the addon
+
+Assuming your addon structure currently looks like this:
```

Comment:
No need to assume - this is what we have created. :wink: 
It makes sense to show the structure here imo, but introduce with something like "This is the addon structure you  created". You could also add a link to the file structure section in the reference here.